pytorch_lightning/plugins/io/torch_plugin.py

```diff
@@ -31,7 +31,10 @@ class TorchCheckpointIO(CheckpointIO):
 
     def save_checkpoint(self, checkpoint: Dict[str, Any], path: _PATH, storage_options: Optional[Any] = None) -> None:
         fs = get_filesystem(path)
-        fs.makedirs(os.path.dirname(path), exist_ok=True)
+        dirname = os.path.dirname(path)
+        if fs.isdir(dirname) and len(fs.ls(dirname)) > 0:
+            rank_zero_warn(f"Checkpoint directory {dirname} exists and is not empty.")
```

Comment:
this will be noisy in the middle of the run. every time we go to save a new checkpoint this warning will show up even though it's expected. perhaps this warning can sit either inside of `__resolve_ckpt_dir`  or in `on_pretraine_routine_start` to indicate that no checkpoints have been saved yet but the path is not empty.

Reply:
oh duh sorry for missing this - maybe move the warning inside this `if` in `__resolve_ckpt_dir`? https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/callbacks/model_checkpoint.py#L619-L620

(commented on maybe keeping this [here](https://github.com/PyTorchLightning/pytorch-lightning/pull/9433#discussion_r712715429))

Reply:
We could use a warning cache to prevent over spamming the users.